vcd/resource_vcd_vapp_network.go

```diff
@@ -0,0 +1,215 @@
+		Delete: resourceVappNetworkDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vapp_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vdc": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"netmask": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "255.255.255.0",
+			},
+			"gateway": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns1": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "8.8.8.8",
+			},
+
+			"dns2": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "8.8.4.4",
+			},
+
+			"dns_suffix": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"guest_vlan_allowed": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+
+			"static_ip_pool": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"end_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceVcdNetworkIPAddressHash,
+			},
+		},
+	}
+}
+
+func resourceVcdVappNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.Mutex.Lock()
+	defer vcdClient.Mutex.Unlock()
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vapp, err := vdc.FindVAppByName(d.Get("vapp_name").(string))
+	if err != nil {
+		return fmt.Errorf("error finding vapp. %#v", err)
+	}
+
+	ipRanges := expandIPRange(d.Get("static_ip_pool").(*schema.Set).List())
+
+	task, err := vapp.AddIsolatedNetwork(&govcd.VappNetworkSettings{
+		Name:             d.Get("name").(string),
+		Gateway:          d.Get("gateway").(string),
+		NetMask:          d.Get("netmask").(string),
+		DNS1:             d.Get("dns1").(string),
+		DNS2:             d.Get("dns2").(string),
+		DNSSuffix:        d.Get("dns_suffix").(string),
+		IPRange:          ipRanges.IPRange,
+		GuestVLANAllowed: d.Get("guest_vlan_allowed").(bool)})
+	if err != nil {
+		return fmt.Errorf("error creating vapp network. %#v", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error waiting from task to complete: %+v", err)
```

Comment:
"waiting from task" -> "waiting for task"

Reply:
Fixed